### PR TITLE
fixing bunch of formatting warnings

### LINKED ISSUE #206: Shrink number of GCC Warnings

### PR DESCRIPTION
**Pull request by LAhmos**
fixed most of the formatting related warnings

### CODE REVIEW

File: src/cuda-sim/instructions.cc [hunk 1]
```diff
@@ -199,7 +199,7 @@ void ptx_thread_info::print_reg_thread(char *fname) {
       const std::string &name = it->first->name();
       const std::string &dec = it->first->decl_location();
       unsigned size = it->first->get_size_in_bytes();
-      fprintf(fp, "%s %llu %s %d\n", name.c_str(), it->second, dec.c_str(),
+      fprintf(fp, "%s %u %s %d\n", name.c_str(), it->second, dec.c_str(),
```
Comment (FJShen) [inline src/cuda-sim/instructions.cc]: Are we certain `it->second` always holds a variable no larger than `unsigned` in this context? If yes, this fix is justifiable. If no, the fix will cause any variable the union is holding - longer than type `unsigned` - to be stripped of higher order bits in the printout.

File: src/cuda-sim/ptx_sim.cc [hunk 2]
```diff
@@ -369,7 +369,7 @@ static void print_reg(FILE *fp, std::string name, ptx_reg_t value,
       fprintf(fp, ".u64 %llu [0x%llx]\n", value.u64, value.u64);
       break;
     case F16_TYPE:
-      fprintf(fp, ".f16 %f [0x%04x]\n", value.f16, (unsigned)value.u16);
+      fprintf(fp, ".f16 %f [0x%04x]\n", static_cast<float>(value.f16), (unsigned)value.u16);
```
Comment (FJShen) [inline src/cuda-sim/ptx_sim.cc:372]: Please use method `half::operator float()` to do the conversion.

**Review by FJShen — CHANGES_REQUESTED** [review]

File: src/cuda-sim/ptx_sim.cc (hunk 2 above)
Comment (FJShen) [reply src/cuda-sim/ptx_sim.cc:372]: Please disregard the above comment. Your code is correct, static_cast will call the overloaded float typecast of `half`.

File: src/cuda-sim/instructions.cc (hunk 1 above)
Comment (FJShen) [reply src/cuda-sim/instructions.cc]: Actually if `it->second` is a union, the correct way to access its contents is to specify a member name, e.g. `it->second.u16`

Comment (LAhmos) [reply src/cuda-sim/instructions.cc]: I agree, I am not sure what happens when we access a union without specifying the member.

Comment (LAhmos) [reply src/cuda-sim/instructions.cc]: I didn't mean to push this as a fix, we still have the warning anyway

**Review by FJShen — APPROVED** [review]

**Comment by FJShen** [comment]
Looks good. 

However `void ptx_thread_info::print_reg_thread(char *fname)` needs to be improved to correctly print the intended contents of `union ptx_reg_t`. Maybe open a new issue to keep track of this bug?

**Pull request merged**